include/deal.II/dofs/dof_handler.h

```diff
@@ -449,7 +449,10 @@ class DoFHandler : public Subscriptor
    * Standard constructor, not initializing any data. After constructing an
    * object with this constructor, use initialize() to make a valid
    * DoFHandler.
+   *
+   * @deprecated Use the constructor which expects a Triangulation instead.
    */
+  DEAL_II_DEPRECATED
   DoFHandler();
```

Comment:
I am fine with this change, but I remark that this can possibly affect a considerable number of user projects. @tamiko I have seen this kind of constructor used often in some of your codes, can you have a look here?